src/controllers/auth.controller.ts

```diff
@@ -163,18 +164,33 @@ export const requireAuth = (
         return
       }
 
-      const token = req.cookies.jwt
+      const token = req.cookies.accessToken
+      const refreshToken = req.cookies.refreshToken
 
-      if (!token) {
-        return res.status(401).json({ error: 'User is not authenticated' })
+      if (!token && !refreshToken) {
+        return res
+          .status(401)
+          .json({ error: 'Access Denied. No token provided.' })
       }
 
       try {
         jwt.verify(token, JWT_SECRET)
       } catch (err) {
-        return res
-          .status(401)
-          .json({ error: 'Invalid token, please log in again' })
+        if (!refreshToken) {
+          return res.status(401).send('Access Denied. Please Login again.')
+        }
+
+        try {
+          const decoded = jwt.verify(refreshToken, REFRESH_JWT_SECRET) as {
+            userId: string
+          }
+
+          signAndSetCookie(res, decoded.userId)
```

Comment:
Calling this updates both access and refresh token. But only the access token should be replaced

Reply:
Does this logic need to be implemented the same as in the `/refresh` endpoint?

Reply:
Here's a sample

```
app.post('/token', (req, res) => {
  const refreshToken = req.cookies['refreshToken'];

  if (!refreshToken || !refreshTokens.includes(refreshToken)) {
    return res.sendStatus(403);
  }

  jwt.verify(refreshToken, refreshTokenSecret, (err, user) => {
    if (err) {
      return res.sendStatus(403);
    }

    const newAccessToken = jwt.sign({ id: user.id, username: user.username }, accessTokenSecret, { expiresIn: '15m' });

    // Update accessToken cookie
    res.cookie('accessToken', newAccessToken, {
      httpOnly: true,
      secure: process.env.NODE_ENV === 'production',
      sameSite: 'strict',
      maxAge: 15 * 60 * 1000 // 15 minutes
    });

    res.json({ message: 'Access token refreshed!' });
  });
});

```

Reply:
resolved

Reply:
resolved